From af0c5ff3abd500258735a986d711180add79958c Mon Sep 17 00:00:00 2001
From: Christian Hewitt <christianshewitt@gmail.com>
Date: Fri, 19 Jun 2020 10:56:46 +0000
Subject: [PATCH 145/152] WIP: arm64: dts: meson: add i2c3/rtc nodes and vrtc
 alias to W400 dtsi

The schematics for all devices currently based on the W400 dtsi show
the same haoyu,hym8563 rtc chip, so add the required i2c3/rtc nodes
and an alias to ensure the meson-vrtc timer which probes earlier is
not assigned the /dev/rtc0 device.

Signed-off-by: Christian Hewitt <christianshewitt@gmail.com>
---
 arch/arm64/boot/dts/amlogic/meson-g12b-w400.dtsi | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/arch/arm64/boot/dts/amlogic/meson-g12b-w400.dtsi b/arch/arm64/boot/dts/amlogic/meson-g12b-w400.dtsi
index 3af3a4db8685..a1c0bf3e8f62 100644
--- a/arch/arm64/boot/dts/amlogic/meson-g12b-w400.dtsi
+++ b/arch/arm64/boot/dts/amlogic/meson-g12b-w400.dtsi
@@ -16,6 +16,7 @@
 	aliases {
 		serial0 = &uart_AO;
 		ethernet0 = &ethmac;
+		rtc1 = &vrtc;
 	};
 
 	chosen {
@@ -301,6 +302,18 @@
 	pinctrl-names = "default";
 };
 
+&i2c3 {
+	status = "okay";
+	pinctrl-0 = <&i2c3_sda_a_pins>, <&i2c3_sck_a_pins>;
+	pinctrl-names = "default";
+
+	rtc@51 {
+		compatible = "haoyu,hym8563";
+		reg = <0x51>;
+		#clock-cells = <0>;
+	};
+};
+
 &pwm_ab {
 	pinctrl-0 = <&pwm_a_e_pins>;
 	pinctrl-names = "default";
-- 
2.17.1

